javatests/arcs/android/integration/policy/PolicyTest.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.android.integration.policy
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.android.integration.IntegrationEnvironment
+import arcs.core.data.RawEntity
+import arcs.core.host.toRegistration
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.core.util.testutil.LogRule
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@OptIn(ExperimentalCoroutinesApi::class, kotlin.time.ExperimentalTime::class)
+@RunWith(AndroidJUnit4::class)
+class PolicyTest {
+
+  @get:Rule
+  val log = LogRule()
+
+  /** Note, a new IntegrationEnvironment is allocated for each test method. */
+  @get:Rule
+  val env = IntegrationEnvironment()
+
+  /**
+   * Scenario: A policy-compliant recipe with @persistent handles egresses data
+   * and stores ingress-restricted values to disk that is never deleted.
+   */
+  @Test
+  fun persistentHandlesWithEgress_StoresIngressRestrictedValues_neverDeleted() = runBlocking {
+    env.addNewHostWith(
+      ::IngressThing.toRegistration(),
+      ::EgressAB.toRegistration()
+    )
+
+    val arc = env.startArc(PersistsEgressesPlan)
+
+    // When the Arc is run...
+    env.waitForIdle(arc)
+
+    val ingest = env.getParticle<IngressThing>(arc)
+    val egressAB = env.getParticle<EgressAB>(arc)
+
+    ingest.storeFinished.join()
```

Comment:
You had a timeout earlier. Is it OK that you just removed it? Or, is there some other API you need to call in the place of a timeout?

Reply:
So, I have `Job`s in the instances of these particle classes. Given that the recipe runs correctly and that `waitForIdle` works, these jobs should be finished. Since this is an async activity that I've added outside of the particle lifecycle, I think it's appropriate to have timeouts. Otherwise, if something goes wrong, the tests will just hang until the runner times out. However, when I remove the timeout block, the tests currently run fine.